src/source/Rtp/Codecs/RtpH264Payloader.c

```diff
@@ -80,27 +80,38 @@ STATUS getNextNaluLength(PBYTE nalus, UINT32 nalusLength, PUINT32 pStart, PUINT3
     ENTERS();
 
     STATUS retStatus = STATUS_SUCCESS;
-    UINT32 zeroCount = 0;
-    PBYTE pCurPtr = nalus;
-    UINT32 i;
+    UINT32 zeroCount = 0, offset;
+    BOOL naluFound = FALSE;
 
     CHK(nalus != NULL && pStart != NULL && pNaluLength != NULL, STATUS_NULL_ARG);
 
     // Annex-B Nalu will have 0x000000001 or 0x000001 start code, at most 4 bytes
-    for (i = 0; i < 4 && i < nalusLength && nalus[i] == 0; i++);
-
-    CHK(i < nalusLength && i < 4 && i >= 2 && nalus[i] == 1, STATUS_RTP_INVALID_NALU);
-    *pStart = ++i;
+    for (offset = 0; offset < 4 && offset < nalusLength && nalus[offset] == 0; offset++);
+
+    CHK(offset < nalusLength && offset < 4 && offset >= 2 && nalus[offset] == 1, STATUS_RTP_INVALID_NALU);
+    *pStart = ++offset;
+
+    while(offset < nalusLength) {
+        if (nalus[offset] == 0) {
+            /* Maybe next byte is 1 */
+            offset++;
```

Comment:
What if you have consequent 0s more than 3 in the row?

Reply:
we are gonna incur a cost there for correctness sake. the next character could be 1.

Reply:
There is a slight perf hit by tracking the count of 0s and checking if the count is 3 and checking one next to be 1. However, I doubt that this if comparison will add any observable perf costs.

Reply:
I dont quite get it, even if we are tracking 0s, we cant step more than 1 if current value is 0. what is the benefit compared to what it is right now?

Reply:
Indeed. You are not bounding your "stepping" say I have 100s of 0s